src/lib.rs

```diff
@@ -348,6 +360,15 @@ macro_rules! try {
         }
     };
 }
+#[cfg(not(feature = "std"))]
+macro_rules! try {
```

Comment:
This duplication is not needed (here, and elsewhere in the PR in imports). It is fine for a normal std crate to import from core. If something is in core, you should avoid duplication by using it from core regardless of what features are set.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=ce95ea2938a54b82d24ff7859882b39a